tests/test_chatbot.py

```diff
@@ -39,61 +39,136 @@ def get_documents(self, source):
+            "completion_kwargs": {
+                "model": "gpt-3.5-turbo",
+            },
+        },
     )
-    chatbot = Chatbot(hf_transformers_cfg)
-    answer = chatbot.process_input("What is a transformer?")
+    buster = Buster(hf_transformers_cfg)
+    answer = buster.process_input("What is a transformer?")
     assert isinstance(answer, str)
 
 
-def test_chatbot_mock_data(tmp_path, monkeypatch):
-    gpt_expected_answer = "this is GPT answer"
-    monkeypatch.setattr("buster.chatbot.get_documents_manager_from_extension", lambda filepath: DocumentsMock)
-    monkeypatch.setattr("buster.chatbot.get_embedding", lambda x, engine: get_fake_embedding())
-    monkeypatch.setattr(
-        "buster.chatbot.openai.Completion.create", lambda **kwargs: {"choices": [{"text": gpt_expected_answer}]}
+def test_chatbot_real_data__chatGPT_OOD():
+    hf_transformers_cfg = BusterConfig(
+        documents_file=DOCUMENTS_FILE,
+        unknown_prompt="I'm sorry, but I am an AI language model trained to assist with questions related to the huggingface transformers library. I cannot answer that question as it is not relevant to the library or its usage. Is there anything else I can assist you with?",
+        embedding_model="text-embedding-ada-002",
+        top_k=3,
+        thresh=0.7,
+        max_words=3000,
+        response_format="slack",
+        completer_cfg={
+            "name": "ChatGPT",
+            "text_before_prompt": (
+                """You are a slack chatbot assistant answering technical questions about huggingface transformers, a library to train transformers in python. """
+                """Make sure to format your answers in Markdown format, including code block and snippets. """
+                """Do not include any links to urls or hyperlinks in your answers. """
+                """If you do not know the answer to a question, or if it is completely irrelevant to the library usage, let the user know you cannot answer. """
+                """For example:\n"""
+                """What is the meaning of life for huggingface?\n"""
+                """This doesn't seem to be related to the huggingface library.\n"""
+                """I'm sorry, but I am an AI language model trained to assist with questions related to the huggingface transformers library. I cannot answer that question as it is not relevant to the library or its usage. Is there anything else I can assist you with?"""
+                """"""
+                """Now answer the following question:\n"""
+            ),
+            "text_before_documents": "Only use these documents as reference:\n",
+            "completion_kwargs": {
+                "model": "gpt-3.5-turbo",
+            },
+        },
     )
+    buster = Buster(hf_transformers_cfg)
+    answer = buster.process_input("What is a good recipe for brocolli soup?")
+    assert isinstance(answer, str)
+    assert hf_transformers_cfg.unknown_prompt in answer
```

Comment:
This sounds super hard. Shouldn't you check the embeddings are close instead?

Reply:
this is what happens in practice already, the check is done within buster. If check passes, the response gets hardcoded to the `unknown_prompt`